ddtrace/internal/telemetry/telemetry_request.py

```diff
@@ -0,0 +1,106 @@
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ..compat import TypedDict
+from ..compat import monotonic
+from ..runtime import get_runtime_id
+from .data import APPLICATION
+from .data import HOST
+
+
+# Contains all the body fields required by v1 of the Telemetry Intake API
+RequestBody = TypedDict(
+    "RequestBody",
+    {
+        "tracer_time": int,
+        "runtime_id": str,
+        "api_version": str,
+        # seq_id (sequence id) should be incremented every time a telemetry request is
+        # sent to the agent. This field will be used to monitor dropped payloads and
+        # reorder requests on the backend.
+        # The value is set in TelemetryWriter.add_event() to ensure seq_id is monotonically increasing
+        "seq_id": Optional[int],
+        "application": Dict,
+        "host": Dict,
+        "payload": Dict,
+        "request_type": str,
+    },
+)
+
+# Contains all the header and body fields required to send a request to v1 of
+# the Telemetry Intake Service
+TelemetryRequest = TypedDict(
+    "TelemetryRequest",
+    {
+        "headers": Dict[str, str],
+        "body": RequestBody,
+    },
+)
+
+
+def _create_telemetry_request(payload, payload_type, seq_id):
+    # type: (Dict, str, Optional[int]) -> TelemetryRequest
+    """
+    Initializes the required fields for a generic Telemetry Intake Request
+
+    :param Payload payload: The payload object sets fields specific to one of the following payload types:
+        app-started, app-closed, app-integrations-changed, and generate-metrics
+
+    :param seq_id int: arg is a counter representing the number of requests sent by the writer
+    """
+    return {
+        "headers": {
+            "Content-type": "application/json",
+            "DD-Telemetry-Request-Type": payload_type,
+            "DD-Telemetry-API-Version": "v1",
+        },
```

Comment:
I wonder if we should keep this as an implementation detail of sending the request in `TelemetryWriter`, we are basically going to keep creating and storing these values for every request, when they are mostly static (other than payload type).